### DISCUSSION TITLE
Do Github App Install Limits Apply to Enterprise Accounts?

### DISCUSSION THREAD

**Discussion by Joshua-Douglas** [community/community#34732] [Enterprise]
The [Github docs](https://docs.github.com/en/developers/apps/getting-started-with-apps/about-apps#about-github-apps) state that an organization can own up to 100 Github Apps. Does anyone know if this limit applies to Github Enterprise? If the limit does apply, then this could be problematic for my organization (and I'm sure that's a sign that our configuration could use some improvement!). 

We are currently using Github Apps to authenticate builds on our CI/CD server (Jenkins). The Github App authentication seems like the best approach for our organization because our builds need access to the Github API, and because our internal security team requires that credentials stored on Jenkins have single repo access. With these restrictions deploy keys won't work because they don't provide API access and machine user PATs won't work (without creating a lot of users) since they provide access to all the repos the user has access to.

As we continue to add repos, we might exceed in the 100 app limit in a year or two. Is this use of Github Apps unusual? I'm curious how large organizations manage their CI/CD credentials...

If the limit is a hard cap, then we might need to begin creating different organizations.

Thank you for your time!

**Comment by coopernetes** [comment]
I'd also like some feedback on this from the GitHub team. 

According to the documentation, 100 GitHub Apps seems to be the hard limit on both accounts, organizations AND enterprises. The enterprise one is surprising.

https://docs.github.com/en/enterprise-cloud@latest/apps/creating-github-apps/registering-a-github-app/registering-a-github-app#registering-a-github-app

Our approach is likely going to be having a bunch of "app hosting orgs" with no repos or workflows just to host our apps (we are planning to use a few thousand). `github-apps-1`, `github-apps-2`, etc. It's very inelegant. Ideally, enterprise customers can ask for limits to be relaxed or expanded similar to how [actions limits can be increased via support](https://docs.github.com/en/actions/reference/limits).
